ErgoScript-Developer-Course/batch-0/ahrnsetido/func-prog-hw-02.md

```diff
@@ -0,0 +1,96 @@
+# Homework 2
+
+## Recap basic ErgoScript and programming
+
+### Example 1
+
+This example contains simple Data Type definitions
+- [x] Long
+- [x] Byte
+
+```scala 
+  val ErgToNanoErg: Long = 1000000000L
+  val tokenPriceErg: Long = 20 * ErgToNanoErg
+  val thisYear: Byte = 2022
+  val thisMonth: Byte = 4
+```
+### Example 2
+
+This example checks if the value in the input box INPUTS(0) is larger than tokenPriceErg (20 Erg), and contains:
+- [x] If else
+- [x] Boolean
+
+```scala 
+  val userBox: Box = INPUTS(0)
+
+  if(userBox.value > tokenPriceErg ){
+    val enoughErg: Boolean = true
```

Comment:
In ErgoScript, you can only have immutable variables, hence they are called "val" instead of "var". So the way you wrote it, you would have to use `enoughErg` only within the scope of the branch of the `if` statement, you couldn't use it outside again.